impeller/renderer/backend/gles/description_gles.cc

```diff
@@ -0,0 +1,159 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "impeller/renderer/backend/gles/description_gles.h"
+
+#include <algorithm>
+#include <cctype>
+#include <iomanip>
+#include <sstream>
+#include <string>
+#include <utility>
+#include <vector>
+
+#include "impeller/base/strings.h"
+#include "impeller/base/validation.h"
+#include "impeller/renderer/backend/gles/proc_table_gles.h"
+
+namespace impeller {
+
+static std::string GetGLString(const ProcTableGLES& gl, GLenum name) {
+  auto str = gl.GetString(name);
+  if (str == nullptr) {
+    return "";
+  }
+  return reinterpret_cast<const char*>(str);
+}
+
+std::optional<bool> DetermineIfES(const std::string& version) {
+  if (HasPrefix(version, "OpenGL ES")) {
+    return true;
+  } else if (HasPrefix(version, "OpenGL")) {
```

Comment:
This case looks redundant with the `else {}` case.

Reply:
So, I am not actually sure if the OpenGL prefix is required. I seemed to think it was. But the Mac desktop implementation does not seem to have it. The else below this was supposed to be `std::nullopt`. Notice how the bool is optional but a value is always returned. I am going to assume the Mac desktop behavior is correct and return a non-optional for now.

Reply:
Done.